docs/node-operators/how-to/upgrade-network.md

```diff
@@ -127,26 +125,42 @@ name = "v0.68.2"
+Once you have performed steps `2.1 Prepare network configuration (all)` and `2.2 Prepare Visor configuration (visor only)`, you don't have to do anything else. The visor will automatically restart the node once the `core` and the `data-node` (if you run one) report they are ready for Protocol Upgrade.
+
+## 2.3 Prepare for an upgrade without Vega Visor (non visor only)
 
-### 3b. Prepare upgrade if not running Visor
-You should do this step only if you are not running Visor.
+You should perform these steps only if you are NOT running Vega Visor.
 
 1. Download the new version of the Vega binary from the [releases page ↗](https://github.com/vegaprotocol/vega/releases)
 2. Unzip the downloaded binary into your file system.
 3. Update your systemd (or any other process manager) to use the new binary.
 4. Reload a systemd service: `systemctl daemon-reload`, or use your own preferred.
-5. Wait for the `upgrade block`, then restart the network with the new binary and config after the `upgrade block` is present.
 
-#### Check if Vega core is ready for restart
-To see when your Vega core node is ready for protocol upgrade, check the Vega-core logs.
+Now you are ready for `Protocol Upgrade`. Please read `3. Execute an upgrade at the agreed block (non visor only)` to know what to do next.
+
+# 3. Execute an upgrade at the agreed block (non visor only)
+
+You should perform these steps only if you are NOT running Vega Visor. If you use Vega Visor, it will perform these steps for you automatically.
+
+**Important:** Smooth execution of the Protocol Upgrade is critical to the Vega Network, and any downtime or disruption must be minimised.
 
-In the Vega core logs, you will see the following messages:
+The below steps cover both use cases: `core` only (marked `a`) and `core + data-node` (marked `b`).
+
+## 3.1 Wait for the `Protocol Upgrade` block
+
+`(a)`: Monitor `/statistics` until `blockHeight` reaches the `upgrade block` and stops increasing
+`(b)`: Monitor `/statistics` of data-node REST endpoint, until both: `blockHeight` from the response body and `x-block-height` response header, both hit `upgrade block`.
+
+**Important:* both `core` and `data-node` will automatically stop processing blocks at the `Protocol Upgrade` block. Both will process any remaining data and prepare for `Protocol Upgrade` by creating a snapshot and network history segment. Depending on your hardware, it might take a couple of seconds or longer. The `core` and the `data-node` process will not exit. Instead, they both will mark themselves as ready for the upgrade. As a Node Operator, you need to check if `core` is ready for restart for `Protocol Upgrade` (`core` waits for `data-node` before marking itself as ready), then you can safely restart the `core` and `data-node` (if you run one) processes.
+
+## 3.2 Verify logs if it is safe to restart services
+
+`(a)` and `(b)`: In the Vega core logs, you will see the following messages:
```

Comment:
```suggestion
:::note 
Both the core and data node will automatically stop processing blocks at the protocol upgrade block. Both will process any remaining data and prepare for the protocol upgrade by creating a snapshot and network history segment. 

Depending on your hardware, it might take 2-3 seconds or longer. The core and data node process will not exit. Instead, they both will mark themselves as ready for the upgrade. As a node operator, you need to check if core is ready to restart for the protocol upgrade. Core waits for data node before marking itself as ready. Then you can safely restart the core and date node (if running) processes.
:::

### 3.2 Verify from logs if it is safe to restart services

`(a)` and `(b)`: In the Vega core logs, you will see the following messages:
```